### PR TITLE
(BOLT-989) Add documentation for wrapping scripts with a task

### PR DESCRIPTION
**Pull request by adreyer**
@melamos

### CODE REVIEW

File: pre-docs/writing_tasks.md [hunk 1]
```diff
@@ -708,6 +708,51 @@ else
...
+
+   script_file="$PT__installdir/script_example/files/myscript.sh"
+   # If this task is going to be run from windows nodes the wrapper must make sure it's exectutable
+   chmod +x $script_file
+   commandline="$script_file $PT_filename $PT_version"
+   # If the stderr output of the script is important redirect it to stdout.
+   ${commandline} 2>&1
+   ```
+
+If you intend to use this task with PE and assign RBAC permissions for it make
```
Comment (donoghuc) [inline pre-docs/writing_tasks.md]: This paragraph is a repeat. It is really important though.

File: pre-docs/writing_tasks.md [hunk 2]
```diff
@@ -708,6 +708,51 @@ else
...
+
+3. Create a small wrapper task that reads environment variables and calls the task.
+
+   ```
+   #!/usr/bin/env bash
+   set -e
+
+   script_file="$PT__installdir/script_example/files/myscript.sh"
+   # If this task is going to be run from windows nodes the wrapper must make sure it's exectutable
+   chmod +x $script_file
```
Comment (donoghuc) [inline pre-docs/writing_tasks.md]: I don't think what you have here is contains any injection vulnerabilities in the parameters that are passed. 

Tested that stderr from script is also visible:
```
#!/usr/bin/env bash
echo 'in myscript'
args=("$@")
echo ${args[0]}
echo ${args[1]}
(>&2 echo "std error from myscript")
```
```
cas@cas-ThinkPad-T460p:~/working_dir/bolt/bolt-modules/boltlib$ bolt task run test::script_wrapper filename='`rm /root/test.txt`' version='(rm /root/test.txt)' -n jhkyqbnchikhkc4.delivery.puppetlabs.net
Started on jhkyqbnchikhkc4.delivery.puppetlabs.net...
Finished on jhkyqbnchikhkc4.delivery.puppetlabs.net:
  in myscript
  `rm
  /root/test.txt`
  std error from myscript
  {
  }
Successful on 1 node: jhkyqbnchikhkc4.delivery.puppetlabs.net
Ran on 1 node in 0.78 seconds
```

File: pre-docs/writing_tasks.md [hunk 3]
```diff
@@ -708,6 +708,51 @@ else
...
 
+If you intend to use this task with PE and assign RBAC permissions for it make
+sure the script safely handles parameters or validate them to prevent shell
+injection vulnerabilities.
+
+### Wrapping an existing script
+
+If the script is not already installed on targets and you do not want to edit
+it, for example if it's a script someone else maintains, you can wrap the
+script as is with it a small task.
```
Comment (melamos) [inline pre-docs/writing_tasks.md]: as is with it?

File: pre-docs/writing_tasks.md [hunk 4]
```diff
@@ -708,6 +708,51 @@ else
...
+
+Given a script that accepts 2 positional args `filename` and `version` called `myscript.sh`:
```
Comment (melamos) [inline pre-docs/writing_tasks.md]:
```suggestion
Given a script, `myscript.sh`, that accepts 2 positional args, `filename` and `version`:
```

File: pre-docs/writing_tasks.md (hunk 2 above)
Comment (donoghuc) [reply pre-docs/writing_tasks.md]: Actually after talking with about it with @sensei-de-loafage he showed me a way to control word splitting such that if there were spaces in the arguments those would not be split on space. 

```
#!/usr/bin/env bash
set -e
script_file="$PT__installdir/script_example/files/myscript.sh"
# If this task is going to be run from windows nodes the wrapper must make sure it's executable
chmod +x $script_file
commandline=("$script_file" "$PT_filename" "$PT_version")
"${commandline[@]}" 2>&
```
Example showing that the whole line is passed as first arg to script even through it has spaces:
```
cas@cas-ThinkPad-T460p:~/working_dir/bolt$ bolt task run test::script_wrapper filename="\`rm -f /root/test.txt\`" version=version -n xne1m3qhr08la58.delivery.puppetlabs.net
Started on xne1m3qhr08la58.delivery.puppetlabs.net...
Finished on xne1m3qhr08la58.delivery.puppetlabs.net:
  in myscript
  `rm -f /root/test.txt`
  version
  std error from myscript
  {
  }
Successful on 1 node: xne1m3qhr08la58.delivery.puppetlabs.net
Ran on 1 node in 1.56 seconds
```

File: pre-docs/writing_tasks.md (hunk 1 above)
Comment (adreyer) [reply pre-docs/writing_tasks.md]: Yeah I put it in both sections intentionally.

File: pre-docs/writing_tasks.md (hunk 3 above)
Comment (adreyer) [reply pre-docs/writing_tasks.md]: `it` doesn't belong. 

I think `you can wrap the script in a small task without modifying it.` is clearer though

**Pull request merged**